### PR TITLE
updated to use long form apache license and add 3rd party licenses

### PR DESCRIPTION
**Pull request by TomBarnes**
Signed-off-by: Tom Barnes <thomas.j.barnes@intel.com>

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,13 +1,292 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (dcmiddle) [inline LICENSE]: Should this appendix be cut? It's instructions on how to attache the short form.

File: LICENSE [hunk 2]
```diff
@@ -1,13 +1,292 @@
-Copyright 2016 Intel Corporation
```
Comment (dcmiddle) [inline LICENSE]: I think this copyright notice should remain 
See around line 189 "... To apply the Apache License to your work, attach the following boilerplate notice, with the fields enclosed by brackets "[]" replaced with your own identifying information. (Don't include the brackets!) ... Copyright [yyyy] [name of copyright owner]"

Comment (TomBarnes) [reply LICENSE]: That is in the section that provides directions on how to apply the license to "your work"; that is, all the other files you've created, not the license itself.  See for example how its handled in this major Apache-sponsored project: https://github.com/apache/spark/blob/master/LICENSE

Comment (TomBarnes) [reply LICENSE]: ... on which was used as an example for this revision

File: LICENSE (hunk 1 above)
Comment (TomBarnes) [reply LICENSE]: see above

Comment (TomBarnes) [reply LICENSE]: ...and also see: http://www.apache.org/dev/apply-license.html#new

File: LICENSE (hunk 2 above)
Comment (dcmiddle) [reply LICENSE]: ok, makes sense

**Pull request merged**